### DISCUSSION TITLE
How to use global variables (Version index 0x54 )

### DISCUSSION THREAD

**Discussion by Barpo** [Louisvdw/dbus-serialbattery#57] [Q&A] [answered]
I made this:

**battery.py:** 
.
.
class Battery(object):

    def __init__(self, port, baud):
        self.port = port
.
.
        self.version = None
        self.vers_index = None
        self.hardware_version = None
        self.soc = None
.
.
**daly.py:**
.
.
    command_alarm = b"\x98"
    command_vers_index = b"\x54"
    command_version = b"\x62"
.
.
    def refresh_data(self):
        result = self.read_soc_data()
        result = result and self.read_vers_index_data()
        result = result and self.read_cell_voltage_range_data()
        result = result and self.read_temperature_range_data()
        result = result and self.read_fed_data()

        return result
.
.
    def read_vers_index_data(self):
        vers_index_data = self.read_serial_data_daly(self.command_vers_index)
        # check if connection success
        if vers_index_data is False:
            return False

        self.vers_index = unpack_from('>h', vers_index_data)
        return True
.
.

**dbushelper.py:**
.
.
        self._dbusservice.add_path('/ProductName', 'DalyBMS (' + self.battery.type + ') v' +
                                   str(DRIVER_VERSION) + DRIVER_SUBVERSION)
        self._dbusservice.add_path('/FirmwareVersion', **self.battery.vers_index**)
        self._dbusservice.add_path('/HardwareVersion', self.battery.hardware_version)
        self._dbusservice.add_path('/Connected', 1)
.
.

My question is, why is "self.battery.vers_index" not seen as global variable? It is shown as '...'(None) at the userinterface. dbushelper does not get the value from refresh_data

**Comment by Louisvdw** [comment]
Are you sure that read_vers_index_data() does get data from the BMS and does not just return False and never reaching your unpack?
You can put some temporary logging in to see if it is reached

**Reply by Barpo** [reply]
def read_vers_index_data(self):
        vers_index_data = self.read_serial_data_daly(self.command_vers_index)
        # check if connection success
        if vers_index_data is False:
            return False

        global self.vers_index 
        self.vers_index = vers_index_data
        logger.info(self.vers_index)
        return True

        self._dbusservice.add_path('/FirmwareVersion', self.battery.vers_index)


@40000000615b443516ce767c INFO:utils:20200306
@40000000615b443718cde82c INFO:utils:20200306
@40000000615b4439175ef2c4 INFO:utils:20200306

**Reply by Louisvdw** [reply, marked as answer]
It seems to get an value and I also cannot see why it should not populate. I suspect it might populate, but just not display corrently in the interface.

**Comment by Lumo1976** [comment]
Hi Louis,

 

 

I have downloaded the driver to USB and for the life of me can’t get it to load to the Cerbo GX..  Any pointers you can give??

 

Thanks in advance..

 

Shaun

**Comment by Louisvdw** [comment]
The Cerbo GX does not work with the auto installer. It's a known issue on that device.
https://github.com/Louisvdw/dbus-serialbattery/wiki/How-to-install
